storage/internal/benchmarks/directory_benchmark.go

```diff
@@ -0,0 +1,294 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"math/rand"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/api/iterator"
+)
+
+// directory benchmark is a benchmark that uploads a randomly generated
+// directory once and then downloads it once. The objects in the directory may
+// be uploaded/downloaded using parallelization.
+//
+// SET UP
+//   - Initialize structs to populate with the benchmark results.
+//   - Select a random API to use to upload/download the object, unless it is
+//     set in the command-line,
+//   - Select a random size for the objects that will be uploaded; this size will
+//     be between two values configured in the command-line.
+//   - Create a directory with 1 to 1000 files of that size and subdirectories.
```

Comment:
This seems like an arbitrary range, and will make some runs too short to give meaningful data. Can we add a command line flag for number of files? Or a flag for total size of the directory?

Reply:
Yes, I plan on adding a flag for size or file number so it can be more deterministic as well. Waiting for feedback on if a mix of file sizes is appropriate/deterministic enough.

Reply:
Done